packages/tpetra/CMakeLists.txt

```diff
@@ -602,60 +602,118 @@ LIST (APPEND ${PACKAGE_NAME}_ETI_LORDS "int")
+    # guarantees long to be at least 32 bits.
+    IF (HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT)
+      GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_DEFAULT OFF)
+    ELSE ()
+      GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_DEFAULT ON)
+    ENDIF ()
+
+    # GO = unsigned is NOT enabled by default.
+    GLOBAL_SET(HAVE_TPETRA_INST_INT_UNSIGNED_DEFAULT OFF)
+    # GO = unsigned long is NOT enabled by default.
+    GLOBAL_SET(HAVE_TPETRA_INST_INT_UNSIGNED_LONG_DEFAULT OFF)
+
+ELSE(Tpetra_ENABLE_DEPRECATED_CODE)
+
+    # Enable Tpetra Global Ordinals LONG LONG default:
+    GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT ON)
+    MESSAGE(STATUS "Tpetra: Tpetra_INST_INT_LONG_LONG is enabled by default.")
+    IF(DEFINED Teuchos_ENABLE_LONG_LONG_INT)
+        IF(NOT Teuchos_ENABLE_LONG_LONG_INT)
+            MESSAGE(STATUS "Tpetra: Tpetra_INST_INT_LONG_LONG is disabled because Teuchos_ENABLE_LONG_LONG_INT is OFF.")
+            MESSAGE(WARNING "Do NOT set Teuchos_ENABLE_LONG_LONG_INT=OFF !  The default is ON.  We will VERY SOON require that this option be ON.  Don't set it explicitly!  Just use the default value.  See #1184: https://github.com/trilinos/Trilinos/issues/1184 and see also #1183: https://github.com/trilinos/Trilinos/issues/1183")
+        ENDIF(NOT Teuchos_ENABLE_LONG_LONG_INT)
+        GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT ${Teuchos_ENABLE_LONG_LONG_INT})
+    ELSE(DEFINED Teuchos_ENABLE_LONG_LONG_INT)
+        GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT ON)
+    ENDIF(DEFINED Teuchos_ENABLE_LONG_LONG_INT)
+
+    IF(DEFINED Tpetra_INST_INT_LONG)
+        IF(Tpetra_INST_INT_LONG)
+            GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT OFF)
+        ENDIF(Tpetra_INST_INT_LONG)
+    ENDIF(DEFINED Tpetra_INST_INT_LONG)
+
+    IF(HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT)
+        GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_DEFAULT OFF)
+    ELSE(HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT)
+        GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_DEFAULT ON)
+    ENDIF(HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT)
+
+    # Enable Tpetra Global Ordinals INT default:
+    GLOBAL_SET(HAVE_TPETRA_INST_INT_INT_DEFAULT OFF)
+
+    # Enable Tpetra Global Ordinals LONG default:
+    GLOBAL_SET(HAVE_TPETRA_INST_INT_LONG_DEFAULT OFF)
+
+    # Enable Tpetra Global Ordinals UNSIGNED default:
+    GLOBAL_SET(HAVE_TPETRA_INST_INT_UNSIGNED_DEFAULT OFF)
+
+    # Enable Tpetra Global Ordinals UNSIGNED LONG default:
+    GLOBAL_SET(HAVE_TPETRA_INST_INT_UNSIGNED_LONG_DEFAULT OFF)
+
+ENDIF()
 
-# GO = unsigned is NOT enabled by default.
-GLOBAL_SET(HAVE_TPETRA_INST_INT_UNSIGNED_DEFAULT OFF)
-# GO = unsigned long is NOT enabled by default.
-GLOBAL_SET(HAVE_TPETRA_INST_INT_UNSIGNED_LONG_DEFAULT OFF)
 
 #
-# Define options for enabling various GlobalOrdinal types
+# Discover what parameters were enabled by the user so we can differentiate what is ON via USER vs. DEFAULTS later.
 #
+IF(NOT Tpetra_ENABLE_DEPRECATED_CODE)
```

Comment:
I did it the way that worked. 

The CMakeLists.txt file handles these in a pretty convoluted way. It sets the defaults up and saves into `HAVE_TPETRA_INST_INT_*_DEFAULT` variables, with the exception that `LONG_LONG` is dependent on what's going on in Teuchos. From there, it sets `HAVE_TPETRA_INST_INT_*` and `Tpetra_INST_INT_*` parameters via a single TriBiTS command that creates and sets based on whether or not `Tpetra_INST_INT_*` is already set (i.e., explicit user parameter) or it uses the default if not set by the user.  Next, `HAVE_TPETRA_INT_*` gets set based on the corresponding `Tpetra_INST_INT_*` value... and these get fed into `Tpetra_ETI_GORDS` which keeps the actual list of GO types...

The sticky thing is without this, it's not straightforward to determine if `LONG_LONG` is enabled from the default setting or if it was enabled from a user option. This matters because if some other type is enabled then the user is requesting that the default should be overridden and we should disable long long.  -but- what if the user explicitly turns off one of the other parameters but doesn't enable long long?

This logic basically says that if any other type is set and turned ON, we should turn long long off.  

I do not intend to change this at this point unless it's incorrect.